src/httpd_jsonapi.c

```diff
@@ -3600,6 +3645,14 @@ jsonapi_init(void)
 	}
     }
 
+  allow_modifying_stored_playlists = cfg_getbool(cfg_getsec(cfg, "library"), "allow_modifying_stored_playlists");
+  default_pl_dir = cfg_getstr(cfg_getsec(cfg, "library"), "default_playlist_directory");
+  if (default_pl_dir == NULL || (default_pl_dir && access(default_pl_dir, W_OK) < 0))
```

Comment:
Only try to access the default directory if `allow_modifying_stored_playlists` is true, only in that case this is an error.